api/README.md

```diff
@@ -1,16 +1,70 @@
+Thrift API
+==========
+These APIs should be used by the clients connecting to the server to store and
+get results.
 
-# Thrift APIs
-These APIs should be used by the clients using the database to store or to get the results. Any new client should only interact with the database through these APIs.
+Any new client should only interact with the database through these APIs.
 
-## Report server API
-The report server API should be used by any client to view or store check results.
-See [report_server.thrift](https://raw.githubusercontent.com/Ericsson/codechecker/master/api/report_server.thrift).
+Available APIs
+--------------
 
-## Authentication system API
-The authentication layer is used for supporting privileged-access only access.
-See [authentication.thrift](https://raw.githubusercontent.com/Ericsson/codechecker/master/thrift_api/authentication.thrift)
+### Report server API
+The report server API should be used by any client to view or store analysis
+results.
 
-## Product management system API
+### Authentication system API
+The authentication layer is used for supporting privileged-access only access
+and permission management.
+
+### Product management system API
 The product management layer is responsible for handling requests about the
-different products and their configuration. See
-[products.thrift](https://raw.githubusercontent.com/Ericsson/codechecker/master/thrift_api/products.thrift)
\ No newline at end of file
+different products and their configuration.
+
+API versioning
+--------------
+
+CodeChecker supports some backward compatibility between API versions.
+
+The API version is automatically encoded into the `POST` request URLs made by
+Thrift in the following format:
+
+    http://example.com:8001/[product-name]/<API-version>/<API-service>
+
+For example:
+
+    https://example.com:8001/Default/v6.0/CodeCheckerService
+
+Executing `make thrift` in the main folder of CodeChecker will automatically
+build each API version to the `build/` dir.
+
+### How to add new API versions
+
+For minor changes, these changes should go into the **existing** API files,
+with appropriate Thrift-based versioning for `struct`s where necessary.
+
+For major changes, a **new** API must be defined. Start by creating a full
+copy of the previous version's API to a new folder, e.g. `v7.0`.
+
+
+> **NOTE!** Make sure the `Authentication` service API contains the
+> `checkAPIVersion` function. Each client is expected to call this function to
+> test if the server accepts its version.
+
+
+ 1. The API handlers' `namespace` should be modified so that the new
```

Comment:
could you please separate this list into 2 todo list and be a bit more specific.
For example.

TODOs at minor version change:
1. update the minor version in `version.py  SUPPORTED_VERSIONS` dict
2. update the minor in js file `API_VERSION` 'const' in `productlist.js`?
what else??

TODOs and major version change:
1. replace the thrift api files and change namespace suffixes to new major version  e.g. codeCheckerAuthentication_v7, update all imports in python clients and servers to the new API version.
3. update the minor version in `version.py  SUPPORTED_VERSIONS` dict
4. update the minor in js file `API_VERSION` 'const' in `productlist.js`?
what else?

Reply:
The first action point for *major* changes are not true as you've written. I'm putting together a more detailed documentation right now.